tasks/main.yml

```diff
@@ -10,6 +10,14 @@
     name: td-agent
     state: "{{ fluentd_package_state }}"
 
+- name: Change group to adm
+  lineinfile:
+    path: /etc/systemd/system/multi-user.target.wants/td-agent.service
+    regexp: '^Group='
+    line: 'Group={ fluentd_group }'
```

Comment:
If you look closely to this unit file
```
[Unit]
Description=td-agent: Fluentd based data collector for Treasure Data
Documentation=https://docs.treasuredata.com/articles/td-agent
After=network-online.target
Wants=network-online.target

[Service]
User=td-agent
Group=adm
LimitNOFILE=65536
Environment=LD_PRELOAD=/opt/td-agent/lib/libjemalloc.so
Environment=GEM_HOME=/opt/td-agent/lib/ruby/gems/2.7.0/
Environment=GEM_PATH=/opt/td-agent/lib/ruby/gems/2.7.0/
Environment=FLUENT_CONF=/etc/td-agent/td-agent.conf
Environment=FLUENT_PLUGIN=/etc/td-agent/plugin
Environment=FLUENT_SOCKET=/var/run/td-agent/td-agent.sock
Environment=TD_AGENT_LOG_FILE=/var/log/td-agent/td-agent.log
Environment=TD_AGENT_OPTIONS=
EnvironmentFile=-/etc/default/td-agent
PIDFile=/var/run/td-agent/td-agent.pid
RuntimeDirectory=td-agent
Type=forking
# XXX: Fix fluentd executables path
ExecStart=/opt/td-agent/bin/fluentd --log $TD_AGENT_LOG_FILE --daemon /var/run/td-agent/td-agent.pid $TD_AGENT_OPTIONS
ExecStop=/bin/kill -TERM ${MAINPID}
ExecReload=/bin/kill -HUP ${MAINPID}
Restart=always
TimeoutStopSec=120

[Install]
WantedBy=multi-user.target
```

you have this mention `EnvironmentFile=-/etc/default/td-agent`
Let's add the group into the `/etc/default/td-agent` file. Would be much more maintainable :)

Reply:
Not possible, it's an escalation privileges

Reply:
Seen with you privately. Let's add `td-agent` user to the group `adm`. This way, we don't have to modify the service file :smiley: